.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,36 @@
+{
+    "build": {
+        "dockerfile": "Dockerfile",
+        "args": {
+            "docker_image_tag_hash": "b5b06ad736cc8f2df227dabf8ff1da1a2f538939"
+        }
+    },
+    "runArgs": [
+        "--cap-add=SYS_PTRACE",
+        "--security-opt",
+        "seccomp=unconfined"
+    ],
+    "customizations": {
+        "vscode": {
+            "extensions": [
+                "eamodio.gitlens",
+                "fredericbonnet.cmake-test-adapter",
+                "GitHub.vscode-github-actions",
+                "GitHub.vscode-pull-request-github",
+                "ms-vscode.cpptools-extension-pack",
+                "redhat.vscode-xml"
+            ]
+        },
```

Comment:
All this file is mainly if you're using `codespaces`, which is still not your case?
If you want, you can simplify it a lot, or simply embed the `b5...39` hash into your `continuousIntegration.yml` with some `echo "DOCKER_IMAGE_TAG_HASH=b5...39" >> "$GITHUB_OUTPUT"`.

Reply:
I think we would want to enable codespaces even though I don't personally use it now....I would like to try it soon.

Reply:
OK, it believe it's a great feature at low cost!